src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,148 @@
+    public String getArguments() {
+        return this.arguments;
+    }
+
+    /**
+     * Parses the ToDo Arguments from the input.
+     *
+     * @return The description of the ToDo task.
+     * @throws DukeException if description is empty.
+     */
+    public String parseToDoArguments() throws DukeException {
+        if (arguments.length() == 0) {
+            throw new DukeException("Hey! The task description cannot be empty!");
+        }
+        return arguments;
+    }
+
+    /**
+     * Parses the find arguments.
+     *
+     * @return The arguments of the find command.
+     * @throws DukeException if argument is empty.
+     */
+    public String parseFindQuery() throws DukeException {
+        if (arguments.length() == 0) {
+            throw new DukeException("Hey! The task description cannot be empty!");
+        }
+        return arguments;
+    }
+
+    /**
+     * Parses the Deadline arguments from the input.
+     *
+     * @return The String Array containing the description as the first element
+     *     and time as the second element.
+     * @throws DukeException if description is empty.
+     */
+    public String[] parseDeadlineArguments() throws DukeException {
+        if (this.arguments.length() == 0) {
+            throw new DukeException("Hey! The task description cannot be empty!");
+        }
+
+        String[] arr = this.arguments.split(" /by ");
+
+        if (arr.length != 2) {
+            throw new DukeException("Hey, the Deadline given is Invalid! "
+                    + "Make sure that you follow this format:\n" + " 'taskDescription /by yyyy-mm-dd hh:mm'");
+        }
+
+        return arr;
+    }
+
+    /**
+     * Parses the Event arguments from the input.
+     *
+     * @return The String array containing the description as the first element,
+     *     start time as the second element and end time as the third element.
+     * @throws DukeException if description is empty.
+     */
+    public String[] parseEventArguments() throws DukeException {
+        if (this.arguments.length() == 0) {
+            throw new DukeException("Hey! The task description cannot be empty!");
+        }
+        String[] arr = this.arguments.split(" /from ");
+        if (arr.length != 2) {
+            throw new DukeException("Hey, the Event given is Invalid!"
+                    + " Make sure that you follow this format:\n"
+                    + " 'eventDescription /from yyyy-mm-dd hh:mm /to yyyy-mm-dd hh:mm'");
+        }
+        String desc = arr[0];
+
+        String[] startEnd = arr[1].split(" /to ");
+        if (startEnd.length != 2) {
+            throw new DukeException("Hey, the Event given is Invalid!"
+                    + " Make sure that you follow this format:\n"
+                    + " 'eventDescription /from yyyy-mm-dd hh:mm /to yyyy-mm-dd hh:mm'");
+        }
+        String start = startEnd[0];
+        String end = startEnd[1];
+
+        return new String[]{desc, start, end};
+    }
+
```

Comment:
Good job, the JavaDoc comments follow the coding standard closely and is clear in explaining the purposes of the functions and variables!